src/NCPoly.jl

```diff
@@ -20,29 +20,41 @@ end
 
 @doc md"""
     dense_poly_type(::Type{T}) where T<:NCRingElement
+    dense_poly_type(::T) where T<:NCRingElement
 
 The type of multivariate polynomials with coefficients of type `T`.
 Falls back to `Generic.NCPoly{T}` respectively `Generic.Poly{T}`.
 
 # Examples
-```julia
-dense_poly_type(typeof(ZZ()))
+```jldoctest; setup = :(using AbstractAlgebra)
+julia> dense_poly_type(elem_type(AbstractAlgebra.ZZ))
+AbstractAlgebra.Generic.Poly{BigInt}
+
+julia> dense_poly_type(AbstractAlgebra.ZZ(1))
+AbstractAlgebra.Generic.Poly{BigInt}
 ```
 """
 dense_poly_type(::Type{T}) where T<:NCRingElement = Generic.NCPoly{T}
+dense_poly_type(::Type{T}) where T = ArgumentError("Type `$T` must be subtype of `NCRingElement`.")
```

Comment:
Don't you mean to *throw* that exception?

Reply:
Also, why is this needed, in contrast to the other `*_type` functions you are editing here? Might be a good idea to add a comment explaining that?